hail/python/hail/linalg/blockmatrix.py

```diff
@@ -2067,24 +2061,18 @@ def _to_bmir(x, block_size):
     if _is_scalar(x):
         return ValueToBlockMatrix(F64(x), [1, 1], block_size)
     else:
-        return ValueToBlockMatrix(_ndarray_to_makearray(x), list(_ndarray_as_2d(x).shape), block_size)
+        return ValueToBlockMatrix(_list_to_makearray(x.flat),
+                                  list(_ndarray_as_2d(x).shape),
+                                  block_size)
 
 
 def _broadcast_to_shape(bmir, result_shape):
     in_index_expr = _broadcast_index_expr(bmir.typ.shape, bmir.typ.is_row_vector)
     return BlockMatrixBroadcast(bmir, in_index_expr, result_shape, bmir.typ.block_size)
 
 
-def _ndarray_to_makearray(ndarray):
-    data = ndarray.tolist()
-
-    # Flatten in the case of 2-D arrays. Would have to be flattened
-    # and reshaped anyway to construct a BlockMatrix
-    if len(ndarray.shape) == 2:
-        data = [x for row in data for x in row]
-
-    data_as_ir = [F64(x) for x in data]
-    return MakeArray(data_as_ir, hl.tarray(hl.tfloat64))
+def _list_to_makearray(l):
+    return MakeArray([F64(x) for x in l], hl.tarray(hl.tfloat64))
```

Comment:
This is going to be quite bad for large arrays.  I think you should use a `hl.literal` here.  It will create a single IR and just store the value which gets serialized as JSON.

Reply:
Sounds good